opentelemetry/src/trace/recruited_tracer.rs

```diff
@@ -0,0 +1,57 @@
+//! Abstracts away details for acquiring a `Tracer` by instrumented libraries.
+use crate::global::BoxedTracer;
+use once_cell::sync::OnceCell;
+use std::fmt::Debug;
+
+/// Holds either a borrowed `BoxedTracer` or a factory that can produce one when
+/// and if needed.
+///
+/// This unifies handling of obtaining a `Tracer` by library code optimizing for
+/// common cases when it will never be needed.
+#[derive(Debug)]
+pub struct RecruitedTracer<'a> {
```

Comment:
`TracerFactory`?
It doesn't always produce new ones though, as might be implied by that name.

`TracerLender`?
Since it does only return borrowed instances?

Reply:
Maybe `BorrowedTracer`?

Reply:
`Obtainer` seems pretty unnatural to me. Maybe `TracerSource`?